docs(databricks): merge databricks_tags additively across config hierarchy levels (#9379)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/databricks-configs.md

† When `table_format` is `iceberg`, `file_format` must be `delta`.

> ‡ `databricks_tags` are applied via `ALTER` statements. Tags cannot be removed via dbt-databricks once applied. To remove tags, use Databricks directly or a post-hook. As of dbt-databricks v1.12.0, `databricks_tags` set at multiple config hierarchy levels [merge additively](#databricks_tags) rather than the lower level fully replacing the higher one.

Reviewer: ```suggestion
‡ `databricks_tags` are applied via `ALTER` statements. Tags cannot be removed via dbt-databricks once applied. To remove tags, use Databricks directly or a post-hook. Starting in `dbt-databricks` v1.12, `databricks_tags` set at multiple config hierarchy levels [merge additively](#databricks_tags) instead of the lower (more specific) level fully replacing the higher one.
```

\* `using_columns` supports all parameter types listed in [Databricks column mask parameters](https://docs.databricks.com/aws/en/sql/language-manual/sql-ref-syntax-ddl-column-mask#parameters).

> † `databricks_tags` are applied via `ALTER` statements. Tags cannot be removed via dbt-databricks once applied. To remove tags, use Databricks directly or a post-hook. As of dbt-databricks v1.12.0, `databricks_tags` set at multiple config hierarchy levels [merge additively](#databricks_tags) rather than the lower level fully replacing the higher one.

Reviewer: ```suggestion
† `databricks_tags` are applied via `ALTER` statements. Tags cannot be removed via dbt-databricks once applied. To remove tags, use Databricks directly or a post-hook. Starting in `dbt-databricks` v1.12, `databricks_tags` set at multiple config hierarchy levels [merge additively](#databricks_tags) instead of the lower (more specific) level fully replacing the higher one.
```

Tags are applied via `ALTER` statements after the materialization is created. Once applied, tags cannot be removed through dbt-databricks configuration changes. To remove tags, you must use Databricks directly or a post-hook.

> :::caution Breaking change in v1.12.0

Reviewer: Can we change this to the following so it's not too alarming/negative?
```suggestion
:::caution Behavior change in v1.12
```

> As of dbt-databricks v1.12.0, `databricks_tags` set at different config hierarchy levels (for example, project-level and model-level) merge additively instead of the lower-level config completely replacing the higher-level one. Keys defined at the lower level override matching keys from the higher level, and any other keys from the higher level are retained. This applies wherever `databricks_tags` is set, including tables, columns, materialized views, and streaming tables.

Reviewer: ```suggestion
Starting in `dbt-databricks` v1.12.0, `databricks_tags` configurations are merged additively across config hierarchy levels (for example, project-level and model-level), rather than having lower-level configs completely replace higher-level ones.

When the same tag key is defined at multiple levels, the lower-level value takes precedence. Tag keys defined only at higher levels are retained.

This behavior applies anywhere `databricks_tags` can be configured, including tables, columns, materialized views, and streaming tables.
```

:::

For example, with the following project-level and model-level configs:

<File name='dbt_project.yml'>

```yaml
models:
  my_project:
    +databricks_tags:
      a: "b"
      c: "project_value"
```

</File>

<File name='models/my_model.sql'>

```sql
{{ config(
    databricks_tags={'c': 'model_value', 'k': 'v'}
) }}
```

</File>

> The model is tagged with `a: b` (kept from the project level), `c: model_value` (the model-level value overrides the project-level `c`), and `k: v` (added at the model level).

Reviewer: ```suggestion
The resulting tags are:

- `a: b` — retained from the project level
- `c: model_value` — the model-level value overrides the project-level `c`
- `k: v` — added at the model level
```
